File: MS.NET/Fundamentals/Platform/Persistence/textiotest.cs
using System;
using System.IO;

static class Program
{
	public static void Main(string[] args)
	{
		if(args.Length > 2)
		{
			float price = float.Parse(args[0]);
			short stock = short.Parse(args[1]);
			string name = args[2].ToUpper();

			TextWriter writer = new StreamWriter(File.Create("product.txt"));
			writer.WriteLine(price);
			writer.WriteLine(stock);
			writer.WriteLine(name);
			writer.Close();
		}
		else
		{
			TextReader reader = new StreamReader(File.OpenRead("product.txt"));
			float price = float.Parse(reader.ReadLine());
			short stock = short.Parse(reader.ReadLine());
			string name = reader.ReadLine();
			reader.Close();
			Console.WriteLine("{0} {1} {2}", price, stock, name);
		}
	}
}

File: Java/Fundamentals/Platform/Persistence/TextIOTest.java
import java.io.*;

class TextIOTest{
	
	public static void main(String[] args) throws Exception{
		if(args.length > 2){
			float price = Float.parseFloat(args[0]);
			short stock = Short.parseShort(args[1]);
			String name = args[2].toUpperCase();
			PrintWriter target = new PrintWriter(
				new OutputStreamWriter(
				new FileOutputStream("product.txt")));
			target.println(price);
			target.println(stock);
			target.println(name);
			target.close();
		}else{
			BufferedReader source = new BufferedReader(
				new InputStreamReader(
				new FileInputStream("product.txt")));
			float price = Float.parseFloat(source.readLine());
			short stock = Short.parseShort(source.readLine());
			String name = source.readLine();
			source.close();
			System.out.printf("%s %s %s%n", price, stock, name);
		}
	}
}

